learning/scripting/c_sharp/c_sharp_features.rst

```diff
@@ -0,0 +1,93 @@
+.. _doc_c_sharp_features:
+
+Features
+============
+
+This page provied an overview over the commonly used features of both C# and Godot
+and how they are used together.
+
+Type Conversion and Casting
+---------------------------
+
+C# is a statically typed language. Therefore you can't do the following:
+
+.. code-block:: csharp
+
+    var mySprite = GetNode("MySprite")
+    mySprite.SetFrame(0)
+
+The method ``GetNode()`` returns a ``Node`` instance.
+You must explicitly convert it to the desired derived type, ``Sprite`` in this case.
+
+For this, you have various options in C#.
+
+**Casting and Type Checking**
+
+Throws ``InvalidCastException`` if the returned node cannot be casted to Sprite.
+You would use it instead of the ``as`` operator if you are pretty sure it won't fail.
+
+.. code-block:: csharp
+
+    Sprite mySprite = (Sprite)GetNode("MySprite");
+    mySprite.SetFrame(0);
+
+**Using the AS operator**
+
+The ``as`` operator returns null if the node cannot be casted to Sprite,
+and for this reason it cannot be used with value types.
+
+.. code-block:: csharp
+
+    Sprite mySprite = GetNode("MySprite") as Sprite;
+    // You should check whether mySprite is null here
+    mySprite.SetFrame(0);
```

Comment:
Hmm, I would just do the null check rather than writing that comment. One-liner: `mySprite?.SetFrame(0);`.